detectors/gcp/gce.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"cloud.google.com/go/compute/metadata"
+
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/standard"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+// GCE collects resource information of GCE computing instances
+type GCE struct{}
+
+// compile time assertion that GCE implements the resource.Detector interface.
+//var _ resource.Detector = (*GCE)(nil) // uncomment it when Detector is published
```

Comment:
Should be released in https://github.com/open-telemetry/opentelemetry-go/releases/tag/v0.9.0